pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -453,3 +456,20 @@ func estimateGracefulTerminationForPods(kubeClient clientset.Interface, ns strin
 	}
 	return estimate, nil
 }
+
+// sortableGroupVersionResources sorts the input set of resources for deletion, and orders pods to always be last.
+// the idea is that the namespace controller will delete all things that spawn pods first in order to reduce the time
+// those controllers spend creating pods only to be told NO in admission and potentially overwhelming cluster especially if they lack rate limiting.
+type sortableGroupVersionResources []unversioned.GroupVersionResource
+
+func (list sortableGroupVersionResources) Len() int {
+	return len(list)
+}
+
+func (list sortableGroupVersionResources) Swap(i, j int) {
+	list[i], list[j] = list[j], list[i]
+}
+
+func (list sortableGroupVersionResources) Less(i, j int) bool {
+	return list[j].Group == "" && list[j].Resource == "pods"
```

Comment:
Assuming that we don't have different versions of the same group and pods being registered in all those versions.
This is currently not the case, but I agree with @lavalamp that we probably need a test for it.